integration-libs/opf/base/root/services/opf-payment-metadata-store.service.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { OpfPaymentMetadata, PaymentProcessingState } from '../model/opf.model';
+
+const initialState = {
+  termsAndConditionsChecked: false,
+  selectedPaymentOptionId: undefined,
+  paymentProcessingState: PaymentProcessingState.NOT_IN_PROGRESS,
```

Comment:
Is this state only in progress or not? If so, maybe it's worth considering using a boolean value here. It would make things a bit simpler instead of creating an enum value.

Reply:
Was thinking maybe we will need some DECLINE in the future or something else, but you've right, for now it could be just boolean. Changed.